detectors/node/opentelemetry-resource-detector-alibaba-cloud/src/detectors/AlibabaCloudEcsDetector.ts

```diff
@@ -114,7 +114,7 @@ class AlibabaCloudEcsDetector implements Detector {
           typeof statusCode !== 'number' ||
           !(statusCode >= 200 && statusCode < 300)
         ) {
-          res.destroy();
+          res.socket.destroy();
```

Comment:
why this change?

Reply:
I saw a cryptic error message and immediately assumed that's because of a changed node API, since the only variable seemed to be node's version.

It actually might be because of `nock` as well, I don't know. I'll revert the change and turn off tests for node@18 for the detector for now. Thanks for questioning this!

Error message:

```
     Uncaught TypeError: The "stream" argument must be an instance of Stream. Received an instance of Socket
      at new NodeError (node:internal/errors:388:5)
      at eos (node:internal/streams/end-of-stream:65:11)
      at IncomingMessage._destroy (node:_http_incoming:234:21)
      at _destroy (node:internal/streams/destroy:109:10)
      at IncomingMessage.destroy (node:internal/streams/destroy:71:5)
      at OverriddenClientRequest.<anonymous> (src/detectors/AlibabaCloudEcsDetector.ts:122:15)
      at Object.onceWrapper (node:events:652:26)
      at OverriddenClientRequest.emit (node:events:537:28)
      at Timeout.respond [as _onTimeout] (node_modules/nock/lib/playback_interceptor.js:307:11)
      at listOnTimeout (node:internal/timers:564:17)
      at processTimers (node:internal/timers:507:7)
```

cc: @legendecas